src/MinimalKafka/KafkaExtensions.cs

```diff
@@ -4,22 +4,36 @@
 using Microsoft.Extensions.DependencyInjection.Extensions;
 using MinimalKafka.Builders;
 using MinimalKafka.Extension;
+using MinimalKafka.Metadata;
 using MinimalKafka.Serializers;
 using MinimalKafka.Stream;
 using System.Text.Json;
+using TopicMetadata = MinimalKafka.Metadata.TopicMetadata;
 
 namespace MinimalKafka;
 
 
 public interface IAddKafkaBuilder : IKafkaConventionBuilder
 {
     IAddKafkaBuilder WithStreamStore(Type streamStoreType);
+    
+    IAddKafkaBuilder WithDefaultTopicOptions();
+    
+    IAddKafkaBuilder WithTopicOptions<TMessage>(Func<Type, string>? namingConvention, TimeSpan? retentionPeriod);
```

Comment:
I needed to define this method on an interface, with extension methods generic didn't work without specifying all the arguments.